docs/data/data-grid/clipboard/clipboard.md

```diff
@@ -0,0 +1,113 @@
+:::
+
+You can paste data from clipboard using the <kbd class="key">Ctrl</kbd>+<kbd class="key">V</kbd> (<kbd class="key">⌘ Command</kbd>+<kbd class="key">V</kbd> on macOS) keyboard shortcut.
+The paste operation only affects cells in the [columns that are `editable`](/x/react-data-grid/editing/#making-a-column-editable).
+
+Same as with editing, you can use `valueParser` to modify the pasted value and `valueSetter` to update the row with new values.
+See [Value parser and value setter](/x/react-data-grid/editing/#value-parser-and-value-setter) editing section for more details.
+
+The behavior of the clipboard paste operation depends on the selection state of the data grid and the data pasted from clipboard.
+The priority is the following, from highest to lowest:
+
+1. If multiple cells are selected (see [Cell selection<span class="plan-premium" title="Premium plan"></span>](/x/react-data-grid/cell-selection/)), the selected cells are updated with the pasted values.
+2. If one or more rows are selected (see [Row selection](/x/react-data-grid/row-selection/)), the selected rows are updated with the pasted values.
+3. If a single cell is selected, the values are pasted starting from the selected cell.
+
+{{"demo": "ClipboardPaste.js", "bg": "inline"}}
+
+### Disable clipboard paste
+
+To disable clipboard paste, set the `disableClipboardPaste` prop to `true`:
+
+```tsx
+<DataGridPremium disableClipboardPaste />
+```
+
+### Persisting pasted data
+
+Clipboard paste uses the same API for persistence as [Editing](/x/react-data-grid/editing/#persistence)—use the `processRowUpdate` prop to persist the updated row in your data source:
+
+```tsx
+processRowUpdate?: (newRow: R, oldRow: R) => Promise<R> | R;
+```
+
+The row will be updated with a value returned by the `processRowUpdate` callback.
+If the callback throws or returns a rejected promise, the row will not be updated.
+
+The demo below shows how to persist the pasted data in the browser's `sessionStorage`.
+
+{{"demo": "ClipboardPastePersistence.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Events
+
+The following events are fired during the clipboard paste operation:
+
+- `clipboardPasteStart` - fired when the clipboard paste operation starts
+- `clipboardPasteEnd` - fired when all row updates from clipboard paste have been processed
+
+For convenience, you can also listen to these events using their respective props:
+
+- `onClipboardPasteStart`
+- `onClipboardPasteEnd`
+
+The demo below shows how to use these events to display a loading indicator while the clipboard paste operation is in progress:
+
+{{"demo": "ClipboardPasteEvents.js", "bg": "inline"}}
+
+### Format of the pasted data
+
+By default, the clipboard paste expects specific format of the clipboard content: the cell values should be separated by a tab (`\t`) character and the rows should be separated by a new line (`\n`) character.
+
+You can use the `unstable_splitClipboardPastedText` prop to support a different format:
```

Comment:
> I'm under the impression that something like delimitClipboardCopiedText instead of clipboardCopyCellDelimiter would be:
More aligned with the unstable_splitClipboardPastedText prop in terms of the interface

Initially, [I wanted to add a callback prop for clipboard data serialization](https://github.com/mui/mui-x/pull/7389#discussion_r1187676782), but this would require deeper changes in the clipboard copy logic since it reuses the `getDataAsCsv` method that returns a string, not raw data.
https://github.com/mui/mui-x/blob/9483d3a90ab4f4da71b05769de20e3ea6e8babee/packages/grid/x-data-grid/src/hooks/features/clipboard/useGridClipboard.ts#L56

We can add it later if necessary, but I would expect that changing the row delimiter (`\n`) to something else is super rare.

For the clipboard import, the main reason I decided to add the `unstable_splitClipboardPastedText` callback is to potentially allow users to support both `,` and `\t` as cell delimiters.

Reply:
> should we cover both aspects (copy and paste delimiters) in this section?

Good idea 👍